tests/test_google_analytics_dynamic_dimension_metric.py

```diff
@@ -0,0 +1,87 @@
+import tap_tester.connections as connections
+import tap_tester.runner as runner
+import tap_tester.menagerie as menagerie
+from base import GoogleAnalyticsBaseTest
+import random
+
+
+class GoogleAnalyticsDynamicFieldselectionTest(GoogleAnalyticsBaseTest):
+    """Ensure running the tap with Audience Technology streams and fields selected results in the replication of more than automatic fields."""
+
+    @staticmethod
+    def name():
+        return "tap_tester_google_analytics_dynamic_dimension_metric_field_test"
+
+    def get_non_selected_fields(self):
+        dimension = {"ga:browser","ga:operatingSystem","ga:flashVersion","ga:javaEnabled","ga:screenColors",
+                     "ga:screenResolution","ga:hostname","ga:date","ga:year","ga:month","ga:hour"}
+        selected_fields = set(random.sample(dimension, 9))  # randomly select 9 fields from dimension set
+        non_selected_fields = dimension - selected_fields
+        return non_selected_fields
```

Comment:
As we discussed, we need generic implementation which will work with all the streams. 
Changes requested:
- include 2 parameters number of dimensions and metrics allowed
- write 2 more functions for random dimension and metric selection
- create class method which will validate the random dimension and metric combination to avoid testing invalid combinations.